webclient/app/src/app/commons/buttons/gitDownloadButton/GitDataButton.jsx

```diff
@@ -4,12 +4,14 @@ import Dialog from "@mui/material/Dialog";
 import DialogActions from "@mui/material/DialogActions";
 import {useTranslation} from "react-i18next";
 import {Close, CloudSync} from "@mui/icons-material";
-import ValidatedTextField from "../validatedTextField/ValidatedTextField";
-import RegexConfig from "../../../regexConfig";
+import ValidatedTextField from "../../inputfields/validatedTextField/ValidatedTextField";
 import GitDataButtonStyles from "./GitDataButtonStyles";
 
 function GitDataButton(props) {
     const {credentialsRequired, handleGit, handleAfterSuccess, buttonIcon, buttonName, buttonVariant} = props;
+    const authUser= /^[a-zA-Z0-9!@#$%^&()*./_-]{2,20}$/;
```

Comment:
Vielleicht könnten wir das in ein großes JSON Objekt auslagern? Dann hätte man alles sehr zentral. Hab ich in meinem Projekt ausprobiert und das funktioniert ganz gut. Hab den Validated Text Field auch mit der Größe erweitert. Dann kann man pro "Eingabe" ein min, max und Pattern hinzufügen.
Könnte die Komponente dann einfach rüber schieben. 
Und eine Anzeige hinzufügen, wie viele Zeichen noch verfügbar sind.

Reply:
authUserRegex oder authUseraPattern wäre hier tatsächlich glaube ich auch besser

Reply:
habe ich umbenannt

Reply:
> Vielleicht könnten wir das in ein großes JSON Objekt auslagern? Dann hätte man alles sehr zentral. Hab ich in meinem Projekt ausprobiert und das funktioniert ganz gut. Hab den Validated Text Field auch mit der Größe erweitert. Dann kann man pro "Eingabe" ein min, max und Pattern hinzufügen. Könnte die Komponente dann einfach rüber schieben. Und eine Anzeige hinzufügen, wie viele Zeichen noch verfügbar sind.

halte ich für nicht consistent mit dem Backend und bricht mit den unhabhängigen Komponenten. Können wir aber gern nochmal drüber reden.

Reply:
Ich verstehe das nicht. Ich lagere etwas in eine zentrale Komponente aus, weil es nicht von mir gepflegt wird, sondern von jemanden, der evtl. kein React kann / nicht programmieren kann. So z.B. Styles und Translation. Dieser Content wird definitiv von Entwicklern gepflegt. Warum hier also eine separate Komponente und bei Styles (wo ich eine Schnittstelle brauche) mische ich das in den Quellcode rein?